File: homeworks/hw3/Models/FigureCollection.java
package homeworks.hw3.Models;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class FigureCollection {

    private List<Figure> figures;

    public FigureCollection() {
        figures = new ArrayList<>();
    }

    public int getCount() {
        return figures.size();
    }

    public List<Figure> getFigures() {
        return figures;
    }

    public void addFigure(Figure figure) {
        figures.add(figure);
    }

    public void removeFigure(int index) {
        figures.remove(index);
    }

    public void updateFigure(int index, Figure figure) {
        figures.set(index, figure);
    }

    public void sortByArea() {
        Collections.sort(figures, new Comparator<Figure>() {
            @Override
            public int compare(Figure f1, Figure f2) {
                return Double.compare(f1.getArea(), f2.getArea());
            }
        });
    }
}

File: finaltask/models/figurecollection.py
from functools import cmp_to_key

class FigureCollection:

    def __init__(self):
        self.figures = []

    def get_count(self):
        return len(self.figures)

    def get_figures(self):
        return self.figures

    def add_figure(self, figure):
        self.figures.append(figure)

    def remove_figure(self, index):
        self.figures.pop(index)

    def update_figure(self, index, figure):
        self.figures[index] = figure

    def sort_by_area(self):
        self.figures.sort(key=cmp_to_key(self.compare_areas))

    @staticmethod
    def compare_areas(f1, f2):
        return (f1.get_area() > f2.get_area()) - (f1.get_area() < f2.get_area())
